cfme/tests/services/test_dialog_regex_validation_in_catalog.py

```diff
@@ -70,6 +76,8 @@ def test_dialog_element_regex_validation(appliance, dialog_cat_item):
     wait_for(lambda: view.submit_button.disabled, timeout=7)
     view.fields(ele_name).fill("test_123")
     wait_for(lambda: not view.submit_button.disabled, timeout=7)
+    view.fields(ele_name).fill("")
+    wait_for(lambda: not view.submit_button.disabled, timeout=7)
```

Comment:
How about using `re` to match the pattern with the string that you use to fill? You can yield the pattern in `dialog_cat_item` along with `catalog_item` and `ele_name`  and then use it to assert if the submit_button is enabled or disabled.
Something like:
```suggestion
    for string in ["!@#$%", "test_123", ""]:
        view.fields(ele_name).fill(string)
        soft_assert(wait_for(
            lambda: view.submit_button.disabled,
            fail_condition=True if (re.match(pattern="^([a-z0-9]+_*)*[a-z0-9]+$", string=string) or string=="") else False,
            timeout=7
        ))
    wait_for(lambda: not view.submit_button.disabled, timeout=7)
```
I would've suggested parametrizing the test with various fill values("!@#$%", "test_123", "") but it will require creating service dialogs, catalogs and catalog items everytime and I don't think it's worth all the time, so you could add a `for` loop. instead.

Reply:
TIL about suggestions, thanks @valaparthvi